ServerSocketTest/src/main/Main.java

```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
+					}
+					else if(resource.equals("/"))//the root doesnt do anything rn. Just a "welcome page"
+					{
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						
+						clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						clientOutput.write(("Hello").getBytes());//encode to bytes
+						
+						clientOutput.flush();//empty the built up buffer
+						
+					}
+					else {
+						//if the resource is not found output 404
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						clientOutput.write(("HTTP/1.1 404 Not Found\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						clientOutput.write(("404").getBytes());//encode to bytes
+						clientOutput.flush();//empty the built up buffer
+					}
+					
 				}
-				else
+				/*if the method is post we need to get the info stored in the body. The body comes after the header. W
+				 * we need to check the content length to know when we are done and we need to check content type so we know 
+				 * which parser to use. currently only Content-Type: application/x-www-form-urlencoded . is supported but need to look into this more*/
+				else if(method.equals("POST"))
 				{
-					OutputStream clientOutput=client.getOutputStream();//output all responses
-					
-					clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
-					clientOutput.write(("\r\n").getBytes());//blank line
-					clientOutput.write(("Hello").getBytes());//encode to bytes
+									
+					if(resource.equals("/people"))
+					{
+						int contentLength=0;
+						//parse out content length using the much beloved regex
+						   // Define the pattern to match "Content-Length: value"
+				        Pattern pattern = Pattern.compile("Content-Length: (\\d+)");
+
+				        // Create a matcher with the input headers
+				        Matcher matcher = pattern.matcher(header);
+
+				        // Check if the pattern is found
+				        if (matcher.find()) {
+				            // Get the matched content length value
+				        	contentLength= Integer.parseInt(matcher.group(1));
+				            System.out.println("Content-Length: " + contentLength);
+				        } else {
+				            System.out.println("Content-Length not found in the headers.");
+				        }
+				        
+						String body="";
+						//System.out.println("h");
+//						int bytesRemaining = 38;
+						
+						//while we still have content to take in to build the body
+						while (contentLength > 0)  {
+							
+						
+							contentLength--;
+							
+							body+=""+(char)buff.read();//check the next char
+							
+							
+						}
+						
+						Pattern pattern2 = Pattern.compile("person\\d=([^&]+)");
+
+				        // Create a matcher with the input string
+				        Matcher matcher2 = pattern2.matcher(body);
```

Comment:
What does the  matcher() class do in this situation?

Reply:
It "counts" the number of matches found in a string (the body)  that correspond to the regex pattern. I should have reassigned pattern and matcher as opposed to create new variables matcher2 partten2 etc

Basically I want to find number of people in the body of the POST and add them to and Array